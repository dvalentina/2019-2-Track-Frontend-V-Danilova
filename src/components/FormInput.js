import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/formInputStyles.module.css';
import { ReactComponent as AttachmentButtonSvg } from '../assets/icons/attachment.svg';
import { ReactComponent as GeoButtonSvg } from '../assets/icons/placeholder.svg';
import { ReactComponent as ImageButtonSvg } from '../assets/icons/picture.svg';
import { ReactComponent as StartRecordButtonSvg } from '../assets/icons/microphone.svg';
import { ReactComponent as StopRecordButtonSvg } from '../assets/icons/pause.svg';
import { ReactComponent as EmojiButtonSvg } from '../assets/icons/emoji.svg';
import EmojiKeyboard from './EmojiKeyboard.js';

export default function FormInput({
	value,
	handleChange,
	handleSubmit,
	handleAttach,
	handleAttachImage,
	handleAttachAudio,
	handleAttachGeolocation,
	isAttachPressed,
	isRecording,
	isEmojiButtonPressed,
	handleEmojiButtonClicked,
	handleEmojiClicked,
}) {
	return (
		<form
			className={styles.formInput} 
			onSubmit={handleSubmit}
			id="send_message"
		>
			<input
				type="text"
				placeholder="Сообщение"
				value={value}
				onChange={handleChange}
				className={styles.inputMessage}
			/>
			<AttachmentButton
				isPressed={isAttachPressed}
				isRecording={isRecording}
				isEmojiButtonPressed={isEmojiButtonPressed}
				handleAttach={handleAttach}
				handleAttachImage={handleAttachImage}
				handleAttachAudio={handleAttachAudio}
				handleAttachGeolocation={handleAttachGeolocation}
				handleEmojiButtonClicked={handleEmojiButtonClicked}
				handleEmojiClicked={handleEmojiClicked}
			/>
		</form>
	);
}

FormInput.propTypes = {
	value: PropTypes.string.isRequired,
	handleChange: PropTypes.func.isRequired,
	handleSubmit: PropTypes.func.isRequired,
	handleAttach: PropTypes.func.isRequired,
	handleAttachImage: PropTypes.func.isRequired,
	handleAttachAudio: PropTypes.func.isRequired,
	handleAttachGeolocation: PropTypes.func.isRequired,
	isAttachPressed: PropTypes.bool.isRequired,
	isRecording: PropTypes.bool.isRequired,
	handleEmojiButtonClicked: PropTypes.func.isRequired,
	isEmojiButtonPressed: PropTypes.bool.isRequired,
	handleEmojiClicked: PropTypes.func.isRequired,
};

function AttachmentButton({
	isPressed,
	isRecording,
	isEmojiButtonPressed,
	handleAttach,
	handleAttachImage,
	handleAttachAudio,
	handleAttachGeolocation,
	handleEmojiButtonClicked,
	handleEmojiClicked,
}) {
	const imageInput = React.useRef(null);
	function handleImageInput() {
		if (imageInput.current) {
			imageInput.current.click();
		}
	}
	if (!isPressed) {
		return (
			<button
				onClick={handleAttach}
				type='button'
				className={styles.attachmentButton}
			>
				<AttachmentButtonSvg className={styles.attachmentButtonSvg} />
			</button>
		);
	}
	return(
		<div className={styles.attachmentButtonsWrap}>
			<EmojiKeyboard
				isPressed={isEmojiButtonPressed}
				handleEmojiClicked={handleEmojiClicked}
			/>
			<button
				onClick={handleEmojiButtonClicked}
				type='button'
				className={styles.attachmentButton}
			>
				<EmojiButtonSvg className={styles.attachmentButtonSvg} />
			</button>
			<button
				onClick={handleImageInput}
				type='button'
				className={styles.attachmentButton}
			>
				<ImageButtonSvg className={styles.attachmentButtonSvg} />
			</button>
			<input
				id='image'
				type='file'
				multiple
				accept='image/*'
				onChange={handleAttachImage}
				ref={imageInput}
				style={{ display: 'none' }}
			/>
			<button
				onClick={handleAttachAudio}
				type='button'
				className={styles.attachmentButton}
			>
				<AudioButtonSvg isRecording={isRecording} />
			</button>
			<button
				onClick={handleAttachGeolocation}
				type='button'
				className={styles.attachmentButton}
			>
				<GeoButtonSvg className={styles.attachmentButtonSvg} />
			</button>
		</div>
	);
}

AttachmentButton.propTypes = {
	handleAttach: PropTypes.func.isRequired,
	handleAttachImage: PropTypes.func.isRequired,
	handleAttachAudio: PropTypes.func.isRequired,
	handleAttachGeolocation: PropTypes.func.isRequired,
	isPressed: PropTypes.bool.isRequired,
	isRecording: PropTypes.bool.isRequired,
	handleEmojiButtonClicked: PropTypes.func.isRequired,
	isEmojiButtonPressed: PropTypes.bool.isRequired,
	handleEmojiClicked: PropTypes.func.isRequired,
};

function AudioButtonSvg({ isRecording }) {
	if (isRecording) {
		return(
			<StopRecordButtonSvg className={styles.attachmentButtonSvg} />
		);
	};
	return(
		<StartRecordButtonSvg className={styles.attachmentButtonSvg} />
	);
}

AudioButtonSvg.propTypes = {
	isRecording: PropTypes.bool.isRequired,
};

export { AudioButtonSvg };
