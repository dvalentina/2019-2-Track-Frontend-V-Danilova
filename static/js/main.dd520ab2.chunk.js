(this.webpackJsonpexam_2=this.webpackJsonpexam_2||[]).push([[0],[,,,,,,,,,,,,function(e,n,a){e.exports={button:"LanguageBlock_button__18dZI",buttonPressed:"LanguageBlock_buttonPressed__2v3oa"}},function(e,n,a){e.exports={languageList:"LanguageList_languageList__2ddmE",languageRow:"LanguageList_languageRow__WY42E"}},function(e,n,a){e.exports={button:"LanguageListButton_button__11K4I",svg:"LanguageListButton_svg__2GqfG"}},function(e,n,a){e.exports={div:"InputBlock_div__1A33b",textarea:"InputBlock_textarea__s1QLP"}},function(e,n,a){e.exports={div:"Translate_div__XqHw9",p:"Translate_p__aSQhN"}},,,,,,,function(e,n,a){e.exports=a.p+"static/media/logo.5d5d9eef.svg"},function(e,n,a){e.exports={div:"LanguageLine_div__8O4sE"}},function(e,n,a){e.exports={div:"TranslateBlockHeader_div__3Gl5q"}},function(e,n,a){e.exports={div:"ResultBlock_div__1MLLV"}},function(e,n,a){e.exports={div:"TranslateBlockBody_div__2Sor-"}},function(e,n,a){e.exports={div:"TranslateBlock_div__39tmB"}},,function(e,n,a){e.exports=a(44)},,,,,function(e,n,a){},function(e,n,a){},,,,,,,,function(e,n,a){"use strict";a.r(n);var t={};a.r(t),a.d(t,"translateText",(function(){return me}));var r=a(0),l=a.n(r),c=a(22),o=a.n(c),u=(a(35),a(17)),s=a(1),i=a(23),g=a.n(i),d=(a(36),a(8)),f=a.n(d),m=a(10),v=a(6),p=a(12),h=a.n(p);function E(e){var n=h.a.button;return!0===e.isPressed&&(n=h.a.buttonPressed),l.a.createElement("button",{className:n,onClick:function(){return e.handleClick(e.name)}},e.name)}var b=a(13),L=a.n(b);function N(e){if(!e.isPressed)return null;var n=[];return e.languageCodes.forEach((function(a,t){n.push(l.a.createElement("button",{className:L.a.languageRow,onClick:function(){return e.handleLanguageClick(t)}},t))})),l.a.createElement("div",{className:L.a.languageList},n)}function _(){return(_=Object.assign||function(e){for(var n=1;n<arguments.length;n++){var a=arguments[n];for(var t in a)Object.prototype.hasOwnProperty.call(a,t)&&(e[t]=a[t])}return e}).apply(this,arguments)}function A(e,n){if(null==e)return{};var a,t,r=function(e,n){if(null==e)return{};var a,t,r={},l=Object.keys(e);for(t=0;t<l.length;t++)a=l[t],n.indexOf(a)>=0||(r[a]=e[a]);return r}(e,n);if(Object.getOwnPropertySymbols){var l=Object.getOwnPropertySymbols(e);for(t=0;t<l.length;t++)a=l[t],n.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(r[a]=e[a])}return r}var x=l.a.createElement("g",null,l.a.createElement("g",null,l.a.createElement("path",{d:"M505.755,123.592c-8.341-8.341-21.824-8.341-30.165,0L256.005,343.176L36.421,123.592c-8.341-8.341-21.824-8.341-30.165,0 s-8.341,21.824,0,30.165l234.667,234.667c4.16,4.16,9.621,6.251,15.083,6.251c5.462,0,10.923-2.091,15.083-6.251l234.667-234.667 C514.096,145.416,514.096,131.933,505.755,123.592z"}))),C=l.a.createElement("g",null),k=l.a.createElement("g",null),w=l.a.createElement("g",null),O=l.a.createElement("g",null),S=l.a.createElement("g",null),y=l.a.createElement("g",null),j=l.a.createElement("g",null),I=l.a.createElement("g",null),R=l.a.createElement("g",null),T=l.a.createElement("g",null),B=l.a.createElement("g",null),P=l.a.createElement("g",null),G=l.a.createElement("g",null),U=l.a.createElement("g",null),H=l.a.createElement("g",null),W=function(e){var n=e.svgRef,a=e.title,t=A(e,["svgRef","title"]);return l.a.createElement("svg",_({id:"Layer_1",x:"0px",y:"0px",viewBox:"0 0 512.011 512.011",style:{enableBackground:"new 0 0 512.011 512.011"},xmlSpace:"preserve",ref:n},t),a?l.a.createElement("title",null,a):null,x,C,k,w,O,S,y,j,I,R,T,B,P,G,U,H)},M=l.a.forwardRef((function(e,n){return l.a.createElement(W,_({svgRef:n},e))})),D=(a.p,a(14)),q=a.n(D);function z(e){var n=Object(r.useState)(!1),a=Object(v.a)(n,2),t=a[0],c=a[1];return l.a.createElement("div",null,l.a.createElement("button",{onClick:function(){c(!t)},type:"button",className:q.a.button},l.a.createElement(M,{className:q.a.svg})),l.a.createElement(N,{handleLanguageClick:function(n){c(!1),e.handleLanguageChange(n),e.handleNewLanguageBlock(n)},languageCodes:e.languageCodes,isPressed:t}))}var J=a(24),K=a.n(J);function Q(e){var n=Object(r.useState)([!1,!1,!1]),a=Object(v.a)(n,2),t=a[0],c=a[1],o=Object(r.useState)(!1),u=Object(v.a)(o,2),s=u[0],i=u[1],g=Object(r.useState)(["ENGLISH","RUSSIAN","GERMAN"]),d=Object(v.a)(g,2),f=d[0],m=d[1],p=null,h=[];function b(n){e.handleLanguageChange(n),N(n)}for(var L=0;L<3;L+=1)h.push(l.a.createElement(E,{name:f[L],isPressed:t[L],handleClick:b}));function N(e){if("DETECT LANGUAGE"===e)i(!0),c([!1,!1,!1]);else{i(!1);var n=[!1,!1,!1];n[f.indexOf(e)]=!0,c(n)}}return"input"===e.type&&(p=l.a.createElement(E,{name:"DETECT LANGUAGE",isPressed:s,handleClick:b})),Object(r.useEffect)((function(){N(e.chosenLanguage)}),[e.chosenLanguage]),l.a.createElement("div",{className:K.a.div},p,h,l.a.createElement(z,{handleNewLanguageBlock:function(e){if(!f.includes(e)){var n=f;n.pop(),n.unshift(e),m(n),h.pop(),h.unshift(l.a.createElement(E,{name:f[0],isPressed:t[0],handleClick:b}))}},languageCodes:e.languageCodes,handleLanguageChange:e.handleLanguageChange}))}var V=a(25),Z=a.n(V);function F(e){return l.a.createElement("div",{className:Z.a.div},l.a.createElement(Q,{languageCodes:e.languageCodes,type:"input",chosenLanguage:e.inputLanguage,handleLanguageChange:e.handleInputLanguageChange}),l.a.createElement(Q,{languageCodes:e.languageCodes,type:"result",chosenLanguage:e.resultLanguage,handleLanguageChange:e.handleResultLanguageChange}))}var X=a(15),Y=a.n(X);function $(e){return l.a.createElement("div",{className:Y.a.div},l.a.createElement("form",null,l.a.createElement("textarea",{className:Y.a.textarea,placeholder:"\u0412\u0432\u0435\u0434\u0438\u0442\u0435 \u0442\u0435\u043a\u0441\u0442 \u0434\u043b\u044f \u043f\u0435\u0440\u0435\u0432\u043e\u0434\u0430",value:e.value,onChange:e.handleChange})))}var ee=a(26),ne=a.n(ee);function ae(e){return l.a.createElement("div",{className:ne.a.div},e.translation)}var te=a(27),re=a.n(te);function le(e){return l.a.createElement("div",{className:re.a.div},l.a.createElement($,{handleChange:e.handleChange,value:e.value}),l.a.createElement(ae,{translation:e.translation}))}var ce=a(28),oe=a.n(ce),ue="trnsl.1.1.20200417T082909Z.013c9f83bfe13d35.92ababf0625c2a9dc3be6cfead958c31f72ae312",se={};function ie(e){return"lang: ".concat(e.lang.slice(-2),", text: ").concat(e.text)}function ge(e,n){var a=ie(e);se[a]=n}function de(e){var n=ie(e),a=se[n];if(a)return a;return{lang:"",text:"",code:404}}var fe=a(38);function me(e){return ve.apply(this,arguments)}function ve(){return(ve=Object(m.a)(f.a.mark((function e(n){var a,t,r;return f.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,de(n);case 2:if(200!==(a=e.sent).code){e.next=6;break}return console.log("next is from cache"),e.abrupt("return",a);case 6:return t=encodeURI("".concat("https://translate.yandex.net/api/v1.5/tr.json/translate","?key=").concat(ue,"&text=").concat(n.text,"&lang=").concat(n.lang)),r=fe(t,{method:"POST"}).then((function(e){return e.json()})).then((function(e){return e})),ge(n,r),e.abrupt("return",r);case 10:case"end":return e.stop()}}),e)})))).apply(this,arguments)}var pe=t;function he(){var e=Object(r.useState)("DETECT LANGUAGE"),n=Object(v.a)(e,2),a=n[0],t=n[1],c=Object(r.useState)("RUSSIAN"),o=Object(v.a)(c,2),u=o[0],s=o[1],i=Object(r.useState)(""),g=Object(v.a)(i,2),d=g[0],p=g[1],h=Object(r.useState)(""),E=Object(v.a)(h,2),b=E[0],L=E[1];var N=new Map([["AZERBAIJAN","az"],["ALBANIAN","sq"],["AMHARIC","am"],["ENGLISH","en"],["ARABIC","ar"],["ARMENIAN","hy"],["BASQUE","eu"],["BASHKIR","ba"],["BELARUSIAN","be"],["BENGALI","bn"],["RUSSIAN","ru"],["GERMAN","de"],["NORWEGIAN","no"],["PERSIAN","fa"],["POLISH","pl"],["SPANISH","es"],["CHINESE","zh"],["KOREAN","ko"],["ITALIAN","it"],["IRISH","ga"]]);function _(){return(_=Object(m.a)(f.a.mark((function e(){var n,t,r,l,c;return f.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return N.has(u)&&(t=N.get(u)),"DETECT LANGUAGE"===a?r="".concat(t):(N.has(a)&&(n=N.get(a)),r="".concat(n,"-").concat(t)),l={lang:r,text:b},e.next=5,pe.translateText(l);case 5:c=e.sent,p(c.text);case 7:case"end":return e.stop()}}),e)})))).apply(this,arguments)}return Object(r.useEffect)((function(){!function(){_.apply(this,arguments)}()})),l.a.createElement("div",{className:oe.a.div},l.a.createElement(F,{languageCodes:N,handleInputLanguageChange:function(e){t(e)},handleResultLanguageChange:function(e){s(e)},inputLanguage:a,resultLanguage:u}),l.a.createElement(le,{translation:d,value:b,handleChange:function(e){L(e.target.value)}}))}var Ee=a(16),be=a.n(Ee);function Le(){return l.a.createElement("div",{className:be.a.div},l.a.createElement("p",{className:be.a.p},"TechnoTrack Translate"),l.a.createElement(he,null))}var Ne=function(){return l.a.createElement(u.a,null,l.a.createElement(s.a,{exact:!0,path:"/"},l.a.createElement("div",{className:"App"},l.a.createElement("header",{className:"App-header"},l.a.createElement("img",{src:g.a,className:"App-logo",alt:"logo"}),l.a.createElement("p",null,"Edit ",l.a.createElement("code",null,"src/App.tsx")," and save to reload."),l.a.createElement("a",{className:"App-link",href:"https://reactjs.org",target:"_blank",rel:"noopener noreferrer"},"Learn React"),l.a.createElement(u.b,{to:"/translate",className:"App-link"},"Translate")))),l.a.createElement(s.a,{path:"/translate"},l.a.createElement(Le,null)))},_e=Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));function Ae(e,n){navigator.serviceWorker.register(e).then((function(e){e.onupdatefound=function(){var a=e.installing;null!=a&&(a.onstatechange=function(){"installed"===a.state&&(navigator.serviceWorker.controller?(console.log("New content is available and will be used when all tabs for this page are closed. See https://bit.ly/CRA-PWA."),n&&n.onUpdate&&n.onUpdate(e)):(console.log("Content is cached for offline use."),n&&n.onSuccess&&n.onSuccess(e)))})}})).catch((function(e){console.error("Error during service worker registration:",e)}))}o.a.render(l.a.createElement(l.a.StrictMode,null,l.a.createElement(Ne,null)),document.getElementById("root")),function(e){if("serviceWorker"in navigator){if(new URL("/2019-2-Track-Frontend-V-Danilova",window.location.href).origin!==window.location.origin)return;window.addEventListener("load",(function(){var n="".concat("/2019-2-Track-Frontend-V-Danilova","/service-worker.js");_e?(!function(e,n){fetch(e,{headers:{"Service-Worker":"script"}}).then((function(a){var t=a.headers.get("content-type");404===a.status||null!=t&&-1===t.indexOf("javascript")?navigator.serviceWorker.ready.then((function(e){e.unregister().then((function(){window.location.reload()}))})):Ae(e,n)})).catch((function(){console.log("No internet connection found. App is running in offline mode.")}))}(n,e),navigator.serviceWorker.ready.then((function(){console.log("This web app is being served cache-first by a service worker. To learn more, visit https://bit.ly/CRA-PWA")}))):Ae(n,e)}))}}()}],[[30,1,2]]]);
//# sourceMappingURL=main.dd520ab2.chunk.js.map