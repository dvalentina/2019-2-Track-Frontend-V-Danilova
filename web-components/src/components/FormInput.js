const template = document.createElement('template');
template.innerHTML = `
  <style>
    input {
      border: 0;
      outline: none;
      width: calc(100% - 2px);
      height: 140px;
      padding-left: 40px;
      padding-right: 40px;
      font-size: 40px;
      display: flex;
      flex-direction: row;
    }

    :host {
      display: inline-block;
      border-top: 1px solid rgba(25, 25, 25, 0.32);
    }

    .button-attachment {
      display: flex;
      fill: grey;
      width: 6vw;
      padding: 35pt;
      align-content: center;
    }
  </style>
  <input type="text">
    <div class="button-attachment">
      <svg version="1.1" id="Capa_1" x="0px" y="0px"
      width="100%" viewBox="0 0 510 510" style="enable-background:new 0 0 510 510;" xml:space="preserve">
        <g>
          <g id="attachment">
            <path d="M140.25,395.25C63.75,395.25,0,331.5,0,255s63.75-140.25,140.25-140.25H408c56.1,0,102,45.9,102,102
              c0,56.1-45.9,102-102,102H191.25c-35.7,0-63.75-28.05-63.75-63.75s28.05-63.75,63.75-63.75H382.5v38.25H191.25
              c-15.3,0-25.5,10.2-25.5,25.5s10.2,25.5,25.5,25.5H408c35.7,0,63.75-28.05,63.75-63.75S443.7,153,408,153H140.25
              c-56.1,0-102,45.9-102,102c0,56.1,45.9,102,102,102H382.5v38.25H140.25z"/>
          </g>
        </g>
      </svg>
    </div>
  </input>
`;

class FormInput extends HTMLElement {
  constructor() {
    super();
    this.shadowRoot = this.attachShadow({ mode: 'open' });
    this.shadowRoot.appendChild(template.content.cloneNode(true));

    this.$input = this.shadowRoot.querySelector('input');
  }

  static get observedAttributes() {
    return ['name', 'value', 'placeholder', 'disabled'];
  }

  attributeChangedCallback(name, oldValue, newValue) {
    this.$input.setAttribute(name, newValue);
  }

  get value() {
    return this.$input.value;
  }

  set value(value) {
    this.$input.value = value;
  }
}

customElements.define('form-input', FormInput);
