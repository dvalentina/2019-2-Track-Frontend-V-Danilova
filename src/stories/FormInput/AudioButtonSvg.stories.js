import React from 'react';
import { AudioButtonSvg } from '../../components/FormInput.js';

export default {
	title: 'Form input audio button svg',
	component: AudioButtonSvg,
};

export const Recording = () => <AudioButtonSvg isRecording />;

export const NotRecording = () => <AudioButtonSvg isRecording = {false} />;
