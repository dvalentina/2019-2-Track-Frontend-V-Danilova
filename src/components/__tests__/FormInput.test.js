import React from 'react';
import renderer from 'react-test-renderer';

import { AudioButtonSvg } from '../FormInput.js';

it('audio button svg with prop isRecording=true renders correctly', () => {
	const audioIsRecording = renderer
		.create(<AudioButtonSvg isRecording />)
		.toJSON();
	expect(audioIsRecording).toMatchSnapshot();
});

it('audio button svg with prop isRecording=false renders correctly', () => {
	const audioIsNotRecording = renderer
		.create(<AudioButtonSvg isRecording={false} />)
		.toJSON();
	expect(audioIsNotRecording).toMatchSnapshot();
});
