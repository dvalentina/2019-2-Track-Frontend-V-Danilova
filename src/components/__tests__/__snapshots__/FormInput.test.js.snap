// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`audio button svg with prop isRecording=false renders correctly 1`] = `
<svg
  className="attachmentButtonSvg"
>
  microphone.svg
</svg>
`;

exports[`audio button svg with prop isRecording=true renders correctly 1`] = `
<svg
  className="attachmentButtonSvg"
>
  pause.svg
</svg>
`;
